src/sentry/integrations/opsgenie/actions/notification.py

```
@@ -0,0 +1,139 @@
+        self.form_fields = {
+            "account": {
+                "type": "choice",
+                "choices": [(i.id, i.name) for i in self.get_integrations()],
+            },
+            "team": {"type": "choice", "choices": self.get_teams()},
+        }
+
+    def _get_team(self, org_integration: RpcOrganizationIntegration | None):
+        if not org_integration:
+            return None
+        teams = org_integration.config.get("team_table")
+        if not teams:
+            return None
+        team_id = self.get_option("team")
+        for team in teams:
+            if team["id"] == team_id:
+                return team
+        return None
+
+    def after(self, event, state):
+        integration = self.get_integration()
+        if not integration:
+            logger.exception("Integration removed, but the rule still refers to it")
+            return
+
+        org_integration = self.get_organization_integration()
+        if not org_integration:
+            org_integration_id = infer_org_integration(
+                integration_id=integration.id, ctx_logger=logger
+            )
+            org_integration = integration_service.get_organization_integration(
+                integration_id=integration.id, organization_id=org_integration_id
+            )
+        team = self._get_team(org_integration)
+        if not team:
+            logger.exception(
+                "The Opsgenie team no longer exists, or the team does not belong to the selected account."
+            )
+            return
+
+        def send_notification(event, futures):
+            org_integration = self.get_organization_integration()
+            org_integration_id = None
+            if org_integration:
+                org_integration_id = org_integration.id
+            else:
+                org_integration_id = infer_org_integration(
+                    integration_id=integration.id, ctx_logger=logger
+                )
```

Same as above, we should raise an error when we can't find the associated org_integration rather than inferring it from the integration. We never want these to fire on behalf of a different organization (even if they share an integration).